### DISCUSSION TITLE
Am I misunderstanding alertmanager config?

### DISCUSSION THREAD

**Discussion by scomtott** [prometheus-operator/kube-prometheus#1119] [Q&A]
The docs (https://github.com/prometheus-operator/kube-prometheus/blob/main/docs/developing-prometheus-rules-and-grafana-dashboards.md#alerting-rules) state:

> According to the Prometheus Monitoring Mixins proposal Prometheus alerting rules are under the key `prometheusAlerts` in the top level object, so in order to add an additional alerting rule, we can simply merge an extra rule into the existing object.

However, the code snippet below makes no mention of `prometheusAlerts` and instead configures `exampleApplication`.

```jsonnet
local kp = (import 'kube-prometheus/main.libsonnet') + {
  values+:: {
    common+: {
      namespace: 'monitoring',
    },
  },
  exampleApplication: {
    prometheusRuleExample: {
      apiVersion: 'monitoring.coreos.com/v1',
      kind: 'PrometheusRule',
      metadata: {
        name: 'my-prometheus-rule',
        namespace: $.values.common.namespace,
      },
      spec: {
        groups: (import 'existingrule.json').groups,
      },
    },
  },
};
```

Attempting to configure the `example.jsonnet` file as per the code snippet did not result in my additional alerting rules being applied to my kube-prometheus deployment (or at least where I'd expect to find them).

I thought about raising an issue but I feel like I might be misunderstanding how something works here.

**Comment by paulfantom** [comment]
Additional rules can be added in few ways:
1) From a rendered json file as in example you quoted (https://github.com/prometheus-operator/kube-prometheus/blob/main/examples/prometheus-additional-rendered-rule-example.jsonnet)
2) Directly specifying alerts in jsonnet (example in https://github.com/prometheus-operator/kube-prometheus/blob/main/examples/prometheus-additional-alert-rule-example.jsonnet)
3) From a mixin (WIP guide in https://github.com/prometheus-operator/kube-prometheus/pull/1107)

The quoted sentence is incorrect after our recent code refactoring. We no longer have top-level and global `prometheusAlerts` nor `prometheusRules` object. Could you create a PR to correct this?
